### ISSUE TITLE
Clients not booting on NFS root image 

### ISSUE THREAD

**Issue by Xpons** [ltsp/ltsp#126]
Hi, we are working long time ago with LTSP5, and now we are trying to move to the 'new' LTSP. With LTSP5 we use a read only NFS ubuntu image and clients mount with a tmpfs overlay upper layer this lower layer filesystem.

With LTSP v20.03 we get an error on 55-initrd-bottom.sh script execution of 'read-only filesystem' and client startup fail (see attached image).  Script on install_ltsp function tries to sync and remove on NFS read only filesystem without mount any overlay fs.
Please, what we are doing wrong? 

On this script I found this comment:
# Make root writable using a tmpfs overlay and install ltsp-init
# @LTSP.CONF: IMAGE OVERLAY

exists some variable 'IMAGE OVERLAY' on LTSP.conf?, I have not found any documentation about this. And seems like in older versions( 19.12-1)  script 55-initrd-bottom.sh and 55-init.sh was responsible to mount the overlay layer but now only exists this comentary header but nothing more.

It's a code bug or a misconfiguration we have?
thank you very much

Xpons
![Captura](https://user-images.githubusercontent.com/63043587/78235084-638a7e00-74d8-11ea-8e76-b8a6427e8839.PNG)

**Comment by alkisg** [comment]
Hi, can you upload your /srv/tftp/ltsp/ltsp.ipxe, and the output of `ltsp info`?

**Comment by Xpons** [comment]
Hi,
yes here are the ltsp info output:

root@LTSP2020:/srv/tftp/ltsp# ltsp info
LTSP version: 20.03.1-1~202003221407~ubuntu18.04.1

SERVER OS: PRETTY_NAME="Ubuntu 18.04.4 LTS"

CHROOTS:
UDS20200306
UDSlocal

VMs:

IMAGES:
x86_64

and attached our ltsp.ipxe:

[ltsp_ipxe.txt](https://github.com/ltsp/ltsp/files/4420880/ltsp_ipxe.txt)

The two images UDS20200306 and UDSlocal are the NFS root images

Thank you
Xpons

**Comment by alkisg** [comment]
Does it boot from x86_64?

**Comment by Xpons** [comment]
Yes, X86_64 it's a squashfs image and works without problem, only both NFS images (UDS20200306 and UDSlocal) fail boot because not  rw overlay over readonly NFS filesystem it's created.

**Comment by alkisg** [comment]
@Xpons, thanks, you're right, the fix for #43 broke chroot-over-NFS booting. Are you able to apply and test the following 3 line patch? You'll need to run `sudo ltsp initrd` after applying it on the server.

```diff
diff --git a/ltsp/client/initrd-bottom/55-initrd-bottom.sh b/ltsp/client/initrd-bottom/55-initrd-bottom.sh
index b31a2a5..cdf8437 100644
--- a/ltsp/client/initrd-bottom/55-initrd-bottom.sh
+++ b/ltsp/client/initrd-bottom/55-initrd-bottom.sh
@@ -44,6 +44,9 @@ initrd_bottom_main() {
         re set_readahead "$rootmnt"
     elif [ ! -d "$rootmnt/proc" ]; then
         die "$rootmnt/proc doesn't exist and ltsp.image wasn't specified"
+    else
+        re mount_img_src "$rootmnt" "$rootmnt" "$tmpfs"
+        re set_readahead "$rootmnt"
     fi
     test -d "$rootmnt/proc" || die "$rootmnt/proc doesn't exist in $_APPLET"
     re install_ltsp
```

**Comment by alkisg** [comment]
Thanks, fix committed in cc407ed41c44f2d5e995e41f4d7685a3f1e257bc, please reopen if it's still an issue.

I'll upload a new 20.04 version in the PPA.
